File: ABC375/b.py
import math

n = int(input())  # 点の数を入力
x = []
y = []

for i in range(n):
    xi, yi = map(int, input().split())  # 点の座標を入力
    x.append(xi)
    y.append(yi)

ans = 0
nx, ny = 0, 0  # 原点からスタート

for i in range(n):
    ans += math.sqrt((x[i] - nx) ** 2 + (y[i] - ny) ** 2)  # 距離を計算
    nx = x[i]
    ny = y[i]

# 最後に原点に戻る距離を計算
ans += math.sqrt(nx ** 2 + ny ** 2)

print(ans)

File: ABC375/b.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
using pint = pair<int,int>;
using pll = pair<long long, long long>;

#define rep(i, n) for (int i = 0; i < (int)(n); i++)
#define rep1(i,n) for (int i = 1; i <= (int)(n); i++)
int inf = 2147483647; // おおよそ2*10^9
ll INF = 9223372036854775807; //おおよそ9*10^18
//ull UINF == おおよそ1.8*10^19
int main(){
    int n;
    cin >> n;
    vector<int> x(n),y(n);
    rep(i,n){
        cin >> x[i] >> y[i];
    }
    double ans = 0;
    int nx = 0, ny = 0;
    rep(i,n){
        ans += sqrt((x[i]-nx)*(x[i]-nx) + (y[i]-ny)*(y[i]-ny));
        nx = x[i];
        ny = y[i];
    }
    ans += sqrt(nx*nx + ny*ny);
    cout << setprecision(15) << ans << endl;
}